mne/channels/tests/test_standard_montage.py

```diff
@@ -73,3 +82,156 @@ def test_standard_superset():
     for key, value in m_1020._get_ch_pos().items():
         if key not in ('O10', 'O9'):
             assert_allclose(c_1005[key], value, atol=1e-4, err_msg=key)
+
+
+def _simulate_artinis_octamon():
+    """Simulate artinis OctaMon channel data from numpy data.
+
+    This is to test data that is imported with missing or incorrect montage
+    info. This data can then be used to test the set_montage function.
+    """
+    np.random.seed(42)
+    data = np.absolute(np.random.normal(size=(16, 100)))
+    ch_names = ['S1_D1 760', 'S1_D1 850', 'S2_D1 760', 'S2_D1 850',
+                'S3_D1 760', 'S3_D1 850', 'S4_D1 760', 'S4_D1 850',
+                'S5_D2 760', 'S5_D2 850', 'S6_D2 760', 'S6_D2 850',
+                'S7_D2 760', 'S7_D2 850', 'S8_D2 760', 'S8_D2 850']
+    ch_types = ['fnirs_cw_amplitude' for _ in ch_names]
+    sfreq = 10.  # Hz
+    info = create_info(ch_names=ch_names, ch_types=ch_types, sfreq=sfreq)
+    for i, ch_name in enumerate(ch_names):
+        info['chs'][i]['loc'][9] = int(ch_name.split(' ')[1])
+    raw = RawArray(data, info)
+
+    return raw
+
+
+def _simulate_artinis_brite23():
+    """Simulate artinis Brite 23 channel data from numpy data.
+
+    This is to test data that is imported with missing or incorrect montage
+    info. This data can then be used to test the set_montage function.
+    """
+    np.random.seed(0)
+    data = np.random.normal(size=(46, 100))
+    sd_names = ['S1_D1', 'S2_D1', 'S3_D1', 'S4_D1', 'S3_D2', 'S4_D2', 'S5_D2',
+                'S4_D3', 'S5_D3', 'S6_D3', 'S5_D4', 'S6_D4', 'S7_D4', 'S6_D5',
+                'S7_D5', 'S8_D5', 'S7_D6', 'S8_D6', 'S9_D6', 'S8_D7', 'S9_D7',
+                'S10_D7', 'S11_D7']
+    ch_names = []
+    ch_types = []
+    for name in sd_names:
+        ch_names.append(name + ' hbo')
+        ch_types.append('hbo')
+        ch_names.append(name + ' hbr')
+        ch_types.append('hbr')
+    sfreq = 10.  # Hz
+    info = create_info(ch_names=ch_names, ch_types=ch_types, sfreq=sfreq)
+    raw = RawArray(data, info)
+
+    return raw
+
+
+def test_set_montage_artinis():
+    """Test that OctaMon and Brite 23 montages are set properly."""
+    # Compare OctaMon and Brite23 to fsaverage
+    montage_octamon = make_standard_montage('artinis-octamon')
+    trans_octamon = compute_native_head_t(montage_octamon)
+    montage_brite23 = make_standard_montage('artinis-brite23')
+    trans_brite = compute_native_head_t(montage_brite23)
+    fif = op.join(op.dirname(__file__), '..', '..', 'data', 'fsaverage',
+                  'fsaverage-trans.fif')
+    fsaverage = read_trans(fif)
+    assert_array_almost_equal(list(trans_octamon.values())[2],
+                              list(fsaverage.values())[2])
+    assert_array_almost_equal(list(trans_brite.values())[2],
+                              list(fsaverage.values())[2])
```

Comment:
Tests failing because compute_native_head_t and fsaverage are different

Reply:
I think we can live with them being not identical, but they should match to some small angle and translation difference. You can get these from the trans matrices, if you're not sure how I can try to push a commit to do it

Reply:
Thanks @larsoner, yes if you have time I would be happy to have a bit of help on it as I'm not very familiar with this.

Reply:
Pushed. I'm surprised the translation is > 1 cm. It doesn't seem like this should happen if they are using the standard locations in MNI coordinates for the fiducials. Do these montages have the following locations for the fiducials?
```
>>> import mne
>>> import pprint
>>> pprint.pprint(mne.coreg.get_mni_fiducials('fsaverage'))
[<DigPoint |        LPA : (-80.6, -29.1, -41.3) mm  : MRI (surface RAS) frame>,
 <DigPoint |     Nasion : (1.5, 85.1, -34.8) mm     : MRI (surface RAS) frame>,
 <DigPoint |        RPA : (84.4, -28.5, -41.3) mm   : MRI (surface RAS) frame>]
```
I wonder if what Artinis defines as LPA/RPA differs from how they were defined for `fsaverage-fiducials.fif`. Maybe we should just make a note that this is probably the case and move on. I think for analyses it should be fine as long as you use the `compute_native_head_t(...)` result as your `trans` in plotting functions rather than using `'fsaverage'`.

Reply:
Hi @larsoner, thank you very much.

The Artinis montages have the following locations for the fiducials:
- For OctaMon: LPA = (-82.58, -20.09, -43.10), Nz = (0.96, 83.56, -48.63) and RPA = (80.25, -19.67, -43.88)
- For Brite 23: LPA = (-81.41, -17.18, -45.56), Nz = (-4.62, 82.33, -45.74) and RPA = (79.66, -18.72, -45.89)

It seems that when using *compute_native_head_t* as `trans` this solves the shifting problem observed with Brite 23. This is what I get now with
```python
fig = mne.viz.plot_alignment(raw.info, show_axes=True,
                             subject='fsaverage', coord_frame='mri',
                             trans=trans, surfaces=['brain', 'head'],
                             fnirs=['channels', 'pairs',
                                    'sources', 'detectors'],
                             dig=True, mri_fiducials=True,
                             subjects_dir=subjects_dir, fig=fig)
```
![image](https://user-images.githubusercontent.com/32393169/115006926-55b85880-9ea1-11eb-808e-eaf4f1a06dca.png)
compared to [the previous comment](https://github.com/mne-tools/mne-python/pull/9141#issuecomment-815970429).

If I understand well, the diamonds are the fsaverage MRI fiducials and the spheres are the Artnins montage fiducials?
Does it mean that the relative positioning of the 3 fiducials is not the same for Artinis and fsaverage, and therefore cannot be solved by a simple translation and rotation of the whole montage?

If the user must not use *fsaverage* for `trans`, is there anything else that should change in the above code for *plot_alignment* (eg. maybe `subject`, `coord_frame` and `mri_fiducials`)?

Reply:
> Does it mean that the relative positioning of the 3 fiducials is not the same for Artinis and fsaverage,

Yes exactly, and given that they are different for two the two Artinis montages, it seems like they did *not* transform these points to MNI space. If they had done that, I don't see why they wouldn't have had the same points for their own two montages.

> and therefore cannot be solved by a simple translation and rotation of the whole montage?

It doesn't mean this. `trans` takes care of the rot+trans we need.

> is there anything else that should change in the above code for plot_alignment (eg. maybe subject, coord_frame and mri_fiducials)?

Nope, what you've done with `trans=compute_native_head_t(mon), subject='fsaverage'` should be enough to get things to align properly, and whether you show `mri_fiducials` and which `coord_frame` you use are just viz options.